posthog/api/action.py

```diff
@@ -146,7 +141,7 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
             actions_list.sort(key=lambda action: action.get('count', action['id']), reverse=True)
         return Response({'results': actions_list})
 
-    def _group_events_to_date(self, date_from: datetime.date, date_to: datetime.date, aggregates: QuerySet, interval:str) -> Dict[datetime.datetime, int]:
+    def _group_events_to_date(self, date_from: datetime.datetime, date_to: datetime.datetime, aggregates: QuerySet, interval:str) -> Dict[datetime.datetime, int]:
```

Comment:
Caught a bug while QA'ing and trying to apply an date interval. Traced it to this function. Now with datetime, the values are precise so the groupby in `dataframe = pd.DataFrame(dataframe.groupby('date').mean(), index=time_index)` is not properly grouping the datetimes because the datetime in the timeindex refers to midnight whereas the datetime in aggregates now carries an exact time.

Reply:
Great debugging. I had seen this bug last night and was still trying to find a way to do this. Should be fixed now.